IFC 4.3/UT_Alignment_12d_1/UT_Alignment_12d_1.ifc

```diff
@@ -0,0 +1,42 @@
+ISO-10303-21;
+HEADER;
+FILE_DESCRIPTION (('ViewDefinition[]'), '2;1');
+FILE_NAME ('UT_Alignment_12d_1.ifc', '2020-12-18T23:28:33', ('ljg'), ('12d Solutions - Ranch'), 'IFC Engine rev 1257', '12d Model 15.0 Alpha 408 [DEVELOP] INT Build 18-12-2020', '');
+FILE_SCHEMA (('IFC4X3_RC2'));
+ENDSEC;
+DATA;
+#1 = IFCPROJECT('2LJesmETv6W9jWXnI$q1Aj', #2, 'UT Alignment 12d 1', '', $, $, $, (#12), #7);
+#2 = IFCOWNERHISTORY(#3, #6, $, .ADDED., $, $, $, 1608294513);
+#3 = IFCPERSONANDORGANIZATION(#4, #5, $);
+#4 = IFCPERSON($, $, 'ljg', $, $, $, $, $);
+#5 = IFCORGANIZATION($, '12d Solutions - Ranch', '12d Solutions - Ranch', $, $);
+#6 = IFCAPPLICATION(#5, '15.0 Alpha 408 [DEVELOP] INT Build 18-12-2020', '12d Model', '12d Model');
+#7 = IFCUNITASSIGNMENT((#8, #9, #10, #11));
+#8 = IFCSIUNIT(*, .LENGTHUNIT., $, .METRE.);
+#9 = IFCSIUNIT(*, .AREAUNIT., $, .SQUARE_METRE.);
+#10 = IFCSIUNIT(*, .VOLUMEUNIT., $, .CUBIC_METRE.);
+#11 = IFCSIUNIT(*, .PLANEANGLEUNIT., $, .RADIAN.);
+#12 = IFCGEOMETRICREPRESENTATIONCONTEXT($, 'Model', 3, 1.E-6, #14, #18);
+#13 = IFCLOCALPLACEMENT($, #14);
+#14 = IFCAXIS2PLACEMENT3D(#15, #16, #17);
+#15 = IFCCARTESIANPOINT((0., 0., 0.));
+#16 = IFCDIRECTION((0., 0., 1.));
+#17 = IFCDIRECTION((1., 0., 0.));
+#18 = IFCDIRECTION((0., 1., 0.));
+#19 = IFCSITE('2iTRVf0N96fejtvVjbiFZR', #2, 'Default Site', 'Description of Default Site', $, #13, $, $, .ELEMENT., $, $, $, $, $);
+#20 = IFCRELAGGREGATES('1YL2QP0e55ivtEE6H_3Mm4', #2, 'ProjectContainer', 'ProjectContainer for Sites', #1, (#19));
+#21 = IFCFACILITY('09_nT1I4r0nup4KFCG4Dt7', #2, 'SA  no VG', 'Description of SA  no VG', $, #13, $, $, .ELEMENT.);
+#22 = IFCRELAGGREGATES('3Edrkwm61CrO1NEpyuWmSU', #2, 'SiteContainer', 'SiteContainer For All Models', #19, (#21));
+#23 = IFCRELCONTAINEDINSPATIALSTRUCTURE('2VLpKdG951IPpTkWgVrw_A', #2, 'Building Container', 'BuildingContainer for Elements', (#24), #21);
+#24 = IFCALIGNMENT('3TsZtwmXbER8Q1PJcHGYg2', $, 'RS01 no VG', '12d Alignment String', 'clothoid', $, $, $, $);
+#25 = IFCRELAGGREGATES('17txXWW0189Rb48wJLlvZl', #2, '', '', #24, (#26));
+#26 = IFCALIGNMENTHORIZONTAL('1aYHKEAr9ANuVigr7mUQoM', #2, 'Horizontal Alignment', '12d Horizontal Alignment', $, $, $, 0., (#28, #30, #32));
```

Comment:
this is double and dedundant information, if the segments are added directly in line#26, which is now the prefered way in RC2, then there should be no IfcRelNests as in line#27.

Reply:
Actually the idea is that IfcRelNests references IfcAlignmentSegment instances that in turn reference the IfcAlignmentHorizontalSegment (through attribute GeometricParameters). The number of IfcAlignmentSegment's in this specific IfcRelNests is the same or less than the number of IfcAlignmentHorizontalSegment's and they are only defined if there is a need for such a IfcAlignmentSegment, for example if there are properties attached to this specific segment collected in one or more property sets.

Reply:
but then there is another error

in line#26 the reference is to 3 instances of IfcAlignmentParameterSegment - a ressource based entity - non-rooted - so this is correct

in line#27 the use of _IfcRelNests_ requires the nesting of rooted (subtype of _IfcRoot_) entities, actually _IfcAlignmentSegment_. So the line#27 is syntactically incorrect, since IfcAlignementHorizontalSegment cannot be used within _IfcRelNests_ as _RelatedObjects_.

Reply:
Exactly, line #27 should reference IfcAlignmentSegment instances if relevant, these IfcAlignmentSegment instances then reference IfcAlignmentHorizontalSegment (through attribute GeometricParameters).